main.cpp

```diff
@@ -199,49 +244,69 @@ void *memalign(size_t a, size_t n) {
             spin_unlock(mag->lock);
             return 0;
         }
-    }
+    //}
     mag->offset = i + m;
     spin_unlock(mag->lock);
     p = MAGIC_ADDR + i;
     ((size_t *)p)[-1] = n;
     return p;
 }
 
-void *malloc(size_t n) {
+void *_malloc(size_t n) {
     return memalign(MAGIC_ALGN, n);
 }
 
 size_t malloc_usable_size(const void *p) {
     return ((const size_t *)p)[-1];
 }
 
-void *calloc(size_t n, size_t z) {
+void *_calloc(size_t n, size_t z) {
     void *p;
-    if ((p = malloc((n *= z)))) {
+    if ((p = _malloc((n *= z)))) {
         memset(p, 0, n);
     }
     return p;
 }
 
-void free(void *p) {
+void _free(void *p) {
     // do nothing
 }
 
-void *realloc(void *p, size_t n) {
+void *_realloc(void *p, size_t n) {
     void *q;
     if (!p) {
-        return malloc(n);
+        return _malloc(n);
     }
     if (!n) {
-        free(p);
+        _free(p);
         return 0;
     }
-    if ((q = malloc(n))) {
+    if ((q = _malloc(n))) {
         memcpy(q, p, ((const size_t *)p)[-1]);
     }
     return q;
 }
 
+#if defined(malloc)
+# undef malloc
+#endif
+#define malloc(x) _malloc(x)
```

Comment:
With something like this, you could *probably* get away with moving these to the top of the file if you pulled the trick

```c
#define malloc(...) _malloc(__VA_ARGS__)
```

That way you could avoid having to rename the symbols above.